exes/tfcctl/src/tfcctl.cpp

```diff
@@ -16,24 +16,54 @@ namespace asio = boost::asio;
 namespace po = boost::program_options;
 namespace ipc = tfc::ipc;
 
+template <typename return_t>
+inline auto create_ipc_signal(asio::io_context& ctx, std::string_view name) -> return_t {
+  if (name.contains("bool")) {
+    return ipc::bool_signal(ctx, name);
+  }
+  if (name.contains("int")) {
+    return ipc::int_signal(ctx, name);
+  }
+  if (name.contains("uint")) {
+    return ipc::uint_signal(ctx, name);
+  }
+  if (name.contains("double")) {
+    return ipc::double_signal(ctx, name);
+  }
+  if (name.contains("string")) {
+    return ipc::string_signal(ctx, name);
+  }
+  if (name.contains("json")) {
+    return ipc::json_signal(ctx, name);
+  }
+  throw std::runtime_error{ fmt::format("invalid_type: {}", name) };
+}
+using any_signal = std::variant<std::monostate,
+                                ipc::bool_signal,
+                                ipc::int_signal,
+                                ipc::uint_signal,
+                                ipc::double_signal,
+                                ipc::string_signal,
+                                ipc::json_signal>;
+
```

Comment:
Later note, move this to ipc.hpp and clean up raw communication builders inside impl.hpp.